src/draw/lv_draw.c

```diff
@@ -154,19 +159,32 @@ void lv_draw_finalize_task_creation(lv_layer_t * layer, lv_draw_task_t * t)
     LV_PROFILER_END;
 }
 
+void lv_draw_wait_for_finish(void)
+{
+#if LV_USE_OS
+    lv_draw_unit_t * u = _draw_info.unit_head;
+    while(u) {
+        if(u->wait_for_finish_cb)
+            u->wait_for_finish_cb(u);
+        u = u->next;
+    }
+#endif
+}
+
 void lv_draw_dispatch(void)
 {
     LV_PROFILER_BEGIN;
-    bool render_running = false;
+    bool task_dispatched = false;
     lv_display_t * disp = lv_display_get_next(NULL);
     while(disp) {
         lv_layer_t * layer = disp->layer_head;
         while(layer) {
             if(lv_draw_dispatch_layer(disp, layer))
-                render_running = true;
+                task_dispatched = true;
             layer = layer->next;
         }
-        if(!render_running) {
+        if(!task_dispatched) {
+            lv_draw_wait_for_finish();
```

Comment:
Please confirm if it works like this:
1. VG-Lite queues 5 tasks
2. The 6th task shall be rendered by SW but it depends on the previous 5
3. Other VG-Lite tasks depend on the 6th SW task so VG-Lite cannot queue more tasks
4. As none of the draw units could dispatch new tasks `task_dispatched` will be `false` and `lv_draw_wait_for_finish` will be called
5. The first 5 draw task will be finished
6. SW can draw the 6th

So VG-Lite can queue the tasks until there is a blocking one.

Reply:
So there are two possible configs:
1. VGLite draw unit only (best performance):
```
#define LV_USE_VGLITE_DRAW_THREAD 1
#define LV_DRAW_SW_DRAW_UNIT_CNT 0
```
E.g. `_draw_info.unit_cnt = 1`
This means that there are no dependency tasks at all - tasks should be queued and as long as the GPU is idle the tasks should be flushed to GPU.
The explanation is [here](https://github.com/lvgl/lvgl/pull/6646/files#r1706509069)

2. If both VGLite draw unit + SW draw unit are available (less performance):
```
#define LV_USE_VGLITE_DRAW_THREAD 1
#define LV_DRAW_SW_DRAW_UNIT_CNT 1
```
E.g. `_draw_info.unit_cnt = 2`
Then is should work as you described.

Reply:
Thanks!